### ISSUE TITLE
[Feature request]: Add transform animation

### ISSUE THREAD

**Issue by CodeDoctorDE** [LinwoodDev/Butterfly#86]
### Describe your feature request!

Currently the transformation is not very fluent. We need some animation!

### Additional context

see https://api.flutter.dev/flutter/widgets/InteractiveViewer-class.html

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by CodeDoctorDE** [comment]
Links:
- https://api.flutter.dev/flutter/physics/FrictionSimulation-class.html
- https://github.com/flutter/flutter/blob/233dd97f1efa005cefd8bfdb824db8295e76dac8/packages/flutter/lib/src/widgets/interactive_viewer.dart#L887

**Comment by CodeDoctorDE** [comment]
We could add something like this:
```dart
class FrictionState {
final DateTime timeStamp;
final double velocityX, velocityY, velocityScale;
}
```

And in the view.dart we could create an AnimationController for all these 3 values.
If timeStamp is not the same in BlocListener, recreate the animateWith.

Calculate done value and use `transform.x + done - animation.value`

**Comment by CodeDoctorDE** [comment]
Fixed in 2.1.0-beta.3 behind an experiment
